replica/replica.go

```diff
@@ -606,37 +612,19 @@ func (r *Replica) processPrepareRemoveDisks(disks []string) ([]PrepareRemoveActi
 			return nil, fmt.Errorf("Wrong disk %v doesn't exist", disk)
 		}
 
-		children := r.diskChildrenMap[disk]
-		// 1) leaf node
-		if children == nil {
-			actions = append(actions, PrepareRemoveAction{
+		parent := r.diskData[disk].Parent
+
+		actions = append(actions,
+			PrepareRemoveAction{
+				Action: OpCoalesce,
+				Source: disk,
```

Comment:
if there is swap between source and target of the output, then, the existing callers of PrepareRemoveDisk need to be modified as well.

Reply:
The output of prepare remove disk was only being called at controller to delete snapshot. Code at controller has been removed to Coalesce and Delete the snapshot.